### PR TITLE
Required database for the CRM

### PR DESCRIPTION
**Pull request by gautam071**
Targets #1208 

### Description
-  Link and Create the required databases

### Checklist:

- [x] I have performed a self-review of my own code.

### Screenshot
image

### CODE REVIEW

File: app/Services/CRMServices.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+use Modules\Prospect\Entities\Prospect;
+
+class CRMServices
+{
+    public function getListData()
+    {
+        return[
+            'clients' =>  Client::all(),
+            'prospectes' => Prospect::all(),
+        ];
```
Comment (GauravGusain98) [inline app/Services/CRMServices.php:15]:
```suggestion
        return [
            'clients' =>  Client::all(),
            'prospectes' => Prospect::all(),
        ];
```

File: database/migrations/2021_12_18_110152_create_clients_crm_details_table.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            $table->unsignedBigInteger('client_contact_persons_id')->nullable();
+            $table->foreign('client_contact_persons_id')->references('id')->on('client_contact_persons');
+            $table->timestamp('last_connected');
+            $table->string('last_interaction')->nullable();
+            $table->string('next_step')->nullable();
+            $table->string('source')->nullable();
+            $table->string('status');
+            $table->string('notes')->nullable();
+            $table->timestamps();
+        });
```
Comment (GauravGusain98) [inline database/migrations/2021_12_18_110152_create_clients_crm_details_table.php:33]:
```suggestion
        Schema::create('client_crm_details', function (Blueprint $table) {
            $table->id();
            $table->unsignedBigInteger('client_id');
            $table->foreign('client_id')->references('id')->on('clients');
            $table->unsignedBigInteger('prospect_id')->nullable();
            $table->foreign('prospect_id')->references('id')->on('prospects');
            $table->unsignedBigInteger('client_address_id')->nullable();
            $table->foreign('client_address_id')->references('id')->on('client_addresses');
            $table->unsignedBigInteger('client_contact_person_id')->nullable();
            $table->foreign('client_contact_person_id')->references('id')->on('client_contact_persons');
            $table->timestamp('last_connected');
            $table->string('last_interaction')->nullable();
            $table->string('next_step')->nullable();
            $table->string('source')->nullable();
            $table->string('status');
            $table->string('notes')->nullable();
            $table->timestamps();
        });
```

File: database/migrations/2021_12_18_110152_create_clients_crm_details_table.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    /**
+     * Reverse the migrations.
+     *
+     * @return void
+     */
+    public function down()
+    {
+        Schema::dropIfExists('clients_crm_details');
```
Comment (GauravGusain98) [inline database/migrations/2021_12_18_110152_create_clients_crm_details_table.php:43]:
```suggestion
        Schema::dropIfExists('client_crm_details');
```

File: resources/views/crm/index.blade.php [hunk 4]
```diff
@@ -2,15 +2,17 @@
...
 <div class="container">
-    <br>
     <div class="d-flex justify-content-start row flex-wrap">
+    <br>
+    <?php foreach ($data['clients'] as $client) : ?>
         <div class="col-md-4">
             <div class="card h-75 mx-4 mt-3 mb-5 ">
-                <a class="card-body no-transition" href="#">
-            	   <br><h2 class="text-center">Clients</h2><br>
+                <a class="card-body no-transition" href="https://portal.test/crm/<?php echo $client->name; ?>">
```
Comment (GauravGusain98) [inline resources/views/crm/index.blade.php:10]: Create route for the url.

Comment (GauravGusain98) [reply resources/views/crm/index.blade.php:10]: Also don't use php echo in laravel we have `{{ }}` in laravel

File: resources/views/crm/index.blade.php [hunk 5]
```diff
@@ -2,15 +2,17 @@
...
+            	   <br><h2 class="text-center"><?php echo $client->name; ?></h2><br>
```
Comment (GauravGusain98) [inline resources/views/crm/index.blade.php:11]: Here as well

File: app/Http/Controllers/CRM/CRMController.php [hunk 6]
```diff
@@ -3,11 +3,31 @@
 namespace App\Http\Controllers\CRM;
 
 use App\Http\Controllers\Controller;
+use App\Services\CRMServices;
 
 class CRMController extends Controller
```
Comment (GauravGusain98) [inline app/Http/Controllers/CRM/CRMController.php:8]:
```suggestion
class CrmController extends Controller
```

Comment (GauravGusain98) [reply app/Http/Controllers/CRM/CRMController.php:8]: Also correct it at other places where it is being used.

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Please check the comments and test the functionality once after making the changes.

File: resources/views/crm/index.blade.php [hunk 7]
```diff
@@ -2,15 +2,17 @@
...
 <div class="container">
-    <br>
     <div class="d-flex justify-content-start row flex-wrap">
+    <br>
+    <?php foreach ($data['prospectes'] as $prospectes) : ?>
         <div class="col-md-4">
             <div class="card h-75 mx-4 mt-3 mb-5 ">
-                <a class="card-body no-transition" href="#">
-            	   <br><h2 class="text-center">Clients</h2><br>
+                <a class="card-body no-transition">
```
Comment (GauravGusain98) [inline resources/views/crm/index.blade.php:10]: Do we not need the href here? If not then its ok.

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Please check the comments

File: resources/views/crm/index.blade.php [hunk 8]
```diff
@@ -2,15 +2,17 @@
...
+            	   <br><h2 class="text-center">{{ $prospectes->name; }}</h2><br>
```
[the remark this answers is not in the document]
Comment (gautam071) [reply resources/views/crm/index.blade.php:11]: The whole UI part is changed in the Next PR so no use in this

File: resources/views/crm/index.blade.php (hunk 7 above)
Comment (gautam071) [reply resources/views/crm/index.blade.php:10]: The whole UI part is changed in the Next PR so no use in this

**Review by GauravGusain98 — APPROVED** [review]
Looks fine

**Pull request merged**